test: Add IPv6 ipsec tests

cathay4t opened the pull request:
With NetworkManager-libreswan 1.2.22+, IPv6 libreswan ipsec tunnel is
supported.

Integration test cases included.

Resolves: https://issues.redhat.com/browse/RHEL-32279

ihuguet commented on tests/integration/ipsec_test.py:
```diff
@@ -616,7 +628,75 @@ def test_ipsec_ipv6_libreswan_client_server(
+    ],
+    ids=["4in6", "6in4"],
+)
+def test_ipsec_ipv6_ipv4_subnet_tunnnel(
```
typo: tunnnel with 3 n's

ihuguet commented on tests/integration/ipsec_test.py:
```diff
@@ -774,24 +855,28 @@ def test_ipsec_dhcpv6_off(
-            leftmodecfgclient: no
             right: {IpsecTestEnv.HOSTB_IPV6_CS}
             rightid: '@hostb.example.org'
+            ipsec-interface: 97
```
Why this change?

BTW, adding ipsec-interface uses a different mode in libreswan (Routing based VPN instead of Policy based VPN). Things that works well with one of them might fail with the other (probably because of bugs in NM-libreswan). Should we do all the tests in both modes maybe?

cathay4t replied:
We can only tell whether autoconf IPv6 is disabled as desired with dedicate ipsec xfrm interface.

ihuguet commented on tests/integration/testlib/env.py:
```diff
@@ -38,3 +38,11 @@ def is_k8s():
     return exec_cmd("rpm -E %{?rhel}".split())[1].strip() == "8"
+
+
+def nm_libreswan_micro_version():
+    return int(
+        exec_cmd("rpm -q NetworkManager-libreswan --qf %{VERSION}".split())[1]
+        .split(".")[-1]
+        .strip()
+    )
```
Just checking the micro version is not enough. When we bump to NM-libreswan 1.24.0, micro < 20 will be true, so the failures in the tests will be masked. Better to do a full version comparison.

cathay4t replied:
Fixed by `nm_libreswan_version_int() < version_str_to_int("1.2.22")`

ihuguet commented on tests/integration/testlib/ipsec.py:
```diff
@@ -450,6 +489,7 @@ def setup_hosta_ip():
             }
         )
 
+    print("HAHA ", desired_state)
```
Leftover from debug?

cathay4t replied:
Removed.

liangwen12year commented on tests/integration/testlib/env.py:
```diff
@@ -38,3 +38,15 @@ def is_k8s():
+
+
+def nm_libreswan_version_int():
+    version_str = exec_cmd(
```
`check=True` ? So that the exception will be raised if there is any.

liangwen12year replied:
Other than the nit above, the patch looks good to me.

cathay4t replied:
Added `check=True`.

liangwen12year commented:
Please resolve the conflict, thanks.

cathay4t committed:
test: Fix incorrect version check on NetworkManager-libreswan

Instead of checking micro version, we should convert version to a
integer to compare using this formula:

    10000 * major_version + 100 * minor_version + micro_version

Introduced `nm_libreswan_version_int()` and `version_str_to_int()` for
this and fixed all version check in ipsec test.

Signed-off-by: Gris Ge <fge@redhat.com>
